service/create/asg_stack.go

```diff
@@ -68,6 +67,38 @@ const (
 	defaultEBSVolumeType = "gp2"
 )
 
+func (s *Service) processASGStack(input asgStackInput) (bool, error) {
```

Comment:
processASGStack can be called from the addFunc and the updateFunc. It handles the check for whether the stack exists.